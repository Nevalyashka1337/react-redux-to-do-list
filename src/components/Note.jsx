import React from 'react'
import propTypes from 'prop-types'


function Note({ text, isCompleted, handleCompleted, id, deleteNote }) {
	let btnClass = 'btn btn-do ';
	if ( isCompleted ) {
		btnClass += 'btn-success'
	} else {
		btnClass += 'btn-outline-success'
	}


	return (
		<div className="row mt-1">
			<div className="col d-flex justify-content-between align-items-stretch">
				<div>
					<button onClick={() => handleCompleted(id)} className={btnClass}><ion-icon name="checkmark"></ion-icon></button>
				</div>
				<div className="border-top border-bottom p-1 px-3 flex-grow-1">{text}</div>
				<div>
					<button onClick={() => deleteNote(id)} className="btn btn-danger btn-del"><ion-icon name="trash"></ion-icon></button>
				</div>
			</div>
		</div>
	)
}

Note.propTypes = {
	text: propTypes.string.isRequired,
	isCompleted: propTypes.bool.isRequired,
	handleCompleted: propTypes.func.isRequired,
	deleteNote: propTypes.func.isRequired
}

export default Note