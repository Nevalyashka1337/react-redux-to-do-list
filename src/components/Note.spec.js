import Note from './Note'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { assert } from 'chai'

describe('Note component', () => {
	test('should button has class "btn-success" when completed', () => {
		const props = {
			text: 'buy milk',
			isCompleted: true,
			handleCompleted: () => {},
			id: new Date().getTime(),
			deleteNote: () => {}
		}
		const wrapper = shallow(<Note {...props}/>)
		const btn = wrapper.find('button.btn.btn-do')
		
		assert.isTrue(btn.hasClass('btn btn-do btn-success'))
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})