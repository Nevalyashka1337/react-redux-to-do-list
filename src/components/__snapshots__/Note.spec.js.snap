// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component should button has class "btn-success" when completed 1`] = `
<div
  className="row mt-1"
>
  <div
    className="col d-flex justify-content-between align-items-stretch"
  >
    <div>
      <button
        className="btn btn-do btn-success"
        onClick={[Function]}
      >
        <ion-icon
          name="checkmark"
        />
      </button>
    </div>
    <div
      className="border-top border-bottom p-1 px-3 flex-grow-1"
    >
      buy milk
    </div>
    <div>
      <button
        className="btn btn-danger btn-del"
        onClick={[Function]}
      >
        <ion-icon
          name="trash"
        />
      </button>
    </div>
  </div>
</div>
`;
